### PR TITLE
Make Time Formats more flexible

### PR DESCRIPTION
**Pull request by InfusOnWoW**
By making the dynamic threshold configurable.

And by using Blizzard's time formatting functions, as proposed by
Elvador in #2173. This patch offers basically the same functionality,
though quite differently presented to the user.

There's now a top-level option to choose between:
* Built-in:            63:42   3:07    10    2.4
* Old Blizzard         2h      3m 7s   10s   2.4
* Modern Blizzard      1h 3m   3m 7s   10s   2.4

All formats support (and default to) increased precision below a
threshold. The default threshold remains at 3s.

The threshold setting is also the off-switch for the additional
precision. Setting it to 0, disables that behaviour. This isn't the most
obvious user interface, but creating a special control for it is too
much work and a additional checkbox destroys the layout.

Fixes: #2707

### CODE REVIEW

**Comment by InfusOnWoW** [comment]
Nightwarden pointed out that my change to precision_types is partly incorrect.

And while testing I noticed a bug.

**Comment by InfusOnWoW** [comment]
Should be all good now.

File: WeakAurasTemplates/TriggerTemplatesData.lua [hunk 1]
```diff
@@ -5922,86 +5930,118 @@ end
...
+  tinsert(templates.race.Orc, { spell = 20572, type = "buff", unit = "player", titleSuffix = L["buff"]});
+  --Cannibalize
+  tinsert(templates.race.Scourge, { spell = 20577, type = "ability", titleSuffix = L["cooldown"]});
+  tinsert(templates.race.Scourge, { spell = 20578, type = "buff", unit = "player", titleSuffix = L["buff"]});
+  -- Will of the Forsaken
+  tinsert(templates.race.Scourge, { spell = 7744, type = "ability", titleSuffix = L["cooldown"]});
+  tinsert(templates.race.Scourge, { spell = 7744, type = "buff", unit = "player", titleSuffix = L["buff"]});
+  -- War Stomp
+  tinsert(templates.race.Tauren, { spell = 20549, type = "ability", titleSuffix = L["cooldown"]});
+  tinsert(templates.race.Tauren, { spell = 20549, type = "buff", titleSuffix = L["buff"]});
```
Comment (mrbuds) [inline WeakAurasTemplates/TriggerTemplatesData.lua:5956]:
```suggestion
  tinsert(templates.race.Tauren, { spell = 20549, type = "debuff", titleSuffix = L["debuff"]});
```

File: WeakAurasTemplates/TriggerTemplatesData.lua [hunk 2]
```diff
@@ -5922,86 +5930,118 @@ end
...
+  tinsert(templates.race.Pandaren, { spell = 107079, type = "buff", titleSuffix = L["buff"]});
+  -- Blood Fury
+  tinsert(templates.race.Orc, { spell = 20572, type = "ability", titleSuffix = L["cooldown"]});
+  tinsert(templates.race.Orc, { spell = 20572, type = "buff", unit = "player", titleSuffix = L["buff"]});
+  --Cannibalize
+  tinsert(templates.race.Scourge, { spell = 20577, type = "ability", titleSuffix = L["cooldown"]});
+  tinsert(templates.race.Scourge, { spell = 20578, type = "buff", unit = "player", titleSuffix = L["buff"]});
+  -- War Stomp
+  tinsert(templates.race.Tauren, { spell = 20549, type = "ability", titleSuffix = L["cooldown"]});
+  tinsert(templates.race.Tauren, { spell = 20549, type = "buff", titleSuffix = L["buff"]});
```
Comment (mrbuds) [inline WeakAurasTemplates/TriggerTemplatesData.lua:5990]:
```suggestion
  tinsert(templates.race.Tauren, { spell = 20549, type = "debuff", titleSuffix = L["debuff"]});
```

File: WeakAurasTemplates/TriggerTemplatesData.lua [hunk 3]
```diff
@@ -5922,86 +5930,118 @@ end
...
+  --Beserking
+  tinsert(templates.race.Troll, { spell = 26297, type = "ability", titleSuffix = L["cooldown"]});
+  tinsert(templates.race.Troll, { spell = 26297, type = "buff", unit = "player", titleSuffix = L["buff"]});
+  -- Arcane Torment
```
Comment (mrbuds) [inline WeakAurasTemplates/TriggerTemplatesData.lua:5994]:
```suggestion
  -- Arcane Torrent
```

File: WeakAurasTemplates/TriggerTemplatesData.lua [hunk 4]
```diff
@@ -5922,86 +5930,118 @@ end
...
+  tinsert(templates.race.BloodElf, { spell = 69179, type = "ability", titleSuffix = L["cooldown"]});
+  tinsert(templates.race.BloodElf, { spell = 69179, type = "buff", titleSuffix = L["buff"]});
```
Comment (mrbuds) [inline WeakAurasTemplates/TriggerTemplatesData.lua:5996]:
```suggestion
```

**Comment by InfusOnWoW** [comment]
Something in this PR breaks Conditions for some of my own auras in some cases, so I need to take another look at it.

**Comment by InfusOnWoW** [comment]
This should be good to go now.

**Pull request merged**